### ISSUE TITLE
Swap space on build nodes

### ISSUE THREAD

**Issue by StuporHero** [Shippable/support#4360]
I am running test kitchen against a Chef cookbook, and one of the pieces of software that one of the recipes in that cookbook installs refuses to install without at least 500 MB of swap space. Below is the error message. I can run this build on my own infrastructure, and it passes. I don't really want to run my own node because I've tried it before, and I couldn't keep up with it, so I'd like you to add swap space to your build nodes if at all possible.

```
Checking Temp space: must be greater than 270 MB.   Actual 92195 MB    Passed
       Checking swap space: 0 MB available, 500 MB required.    Failed <<<<
       
       Some requirement checks failed. You must fulfill these requirements before
       
       continuing with the installation
```

**Comment by StuporHero** [comment]
I'd also rather you just said no than to continue to let this ticket sit here. I have other options.

**Comment by StuporHero** [comment]
If anyone else needs this, I discovered I can create and enable a swap file in the `pre_ci` step like this:
```
build:
  pre_ci: |
    if [ ! -f /var/swapfile ]
    then
      dd if=/dev/zero of=/var/swapfile bs=1024 count=1024000
      chmod 600 /var/swapfile
      mkswap /var/swapfile
      swapon /var/swapfile
    fi
  ci: ...
```

**Comment by trriplejay** [comment]
hey, sorry we let this sit without a response. I'll mark this as a feature request. We may be able to add this as a node pool setting.

**Comment by StuporHero** [comment]
@trriplejay Thanks!

**Comment by manishas** [comment]
@StuporHero do you still need this now that you found a way to work around this? We haven't heard this request from anyone else, so trying to assign a priority to it.

**Comment by StuporHero** [comment]
@manishas I suppose it isn't absolutely necessary, but it would help. I can't really do anything with it until August anyway.

**Comment by manishas** [comment]
Thanks for the clarification. 

Unless you're blocked by this, I'm closing this issue as deferred. If we hear some +1 feedback or you say this is a blocker, happy to bring it back into scope.
